tests/components/elgato/conftest.py

```diff
@@ -48,40 +60,44 @@ def mock_onboarding() -> Generator[None, MagicMock, None]:
 
 
 @pytest.fixture
-def mock_elgato_config_flow() -> Generator[None, MagicMock, None]:
+def mock_elgato_config_flow(device_fixtures: str) -> Generator[None, MagicMock, None]:
     """Return a mocked Elgato client."""
     with patch(
         "homeassistant.components.elgato.config_flow.Elgato", autospec=True
     ) as elgato_mock:
         elgato = elgato_mock.return_value
-        elgato.info.return_value = Info.parse_raw(load_fixture("info.json", DOMAIN))
+        elgato.info.return_value = Info.parse_raw(
+            load_fixture(f"{device_fixtures}/info.json", DOMAIN)
+        )
         yield elgato
 
 
 @pytest.fixture
-def mock_elgato(request: pytest.FixtureRequest) -> Generator[None, MagicMock, None]:
+def mock_elgato(
+    device_fixtures: str, state_variant: str
+) -> Generator[None, MagicMock, None]:
     """Return a mocked Elgato client."""
-    variant = {"state": "temperature", "settings": "temperature"}
-    if hasattr(request, "param") and request.param:
-        variant = request.param
-
     with patch(
         "homeassistant.components.elgato.coordinator.Elgato", autospec=True
     ) as elgato_mock:
         elgato = elgato_mock.return_value
-        elgato.info.return_value = Info.parse_raw(load_fixture("info.json", DOMAIN))
+        elgato.info.return_value = Info.parse_raw(
+            load_fixture(f"{device_fixtures}/info.json", DOMAIN)
```

Comment:
Could you make this json loading happen only once per test session?

Reply:
How would I do that? I can't mark this fixture in session scope.

I could wrap the whole `load_fixture` and smash a `lru_cache` decorator on it (actually we could do that for load_fixture by default). But, the downside is that it will consume memory for all the test time (not sure if that is an issue).

Reply:
The benchmark now:

![image](https://user-images.githubusercontent.com/195327/217274392-d50c7ee0-1940-49a4-911f-9c8271630199.png)

I've applied the following:

```diff
diff --git a/tests/components/elgato/conftest.py b/tests/components/elgato/conftest.py
index fad3c04c10..169c0c23fd 100644
--- a/tests/components/elgato/conftest.py
+++ b/tests/components/elgato/conftest.py
@@ -8,11 +8,26 @@ import pytest
 from homeassistant.components.elgato.const import DOMAIN
 from homeassistant.const import CONF_HOST, CONF_MAC, CONF_PORT
 from homeassistant.core import HomeAssistant
+from functools import cache

 from tests.common import MockConfigEntry, load_fixture
 from tests.components.light.conftest import mock_light_profiles  # noqa: F401


+@cache
+def __load_fixtures(
+    device_fixtures: str, state_variant: str
+) -> tuple[Info, State, Settings]:
+    """Load fixtures for a specific device."""
+    return (
+        Info.parse_raw(load_fixture(f"{device_fixtures}/info.json", DOMAIN)),
+        State.parse_raw(
+            load_fixture(f"{device_fixtures}/{state_variant}.json", DOMAIN)
+        ),
+        Settings.parse_raw(load_fixture(f"{device_fixtures}/settings.json", DOMAIN)),
+    )
+
+
 @pytest.fixture
 def device_fixtures() -> str:
     """Return the device fixtures for a specific device."""
@@ -77,19 +92,14 @@ def mock_elgato(
     device_fixtures: str, state_variant: str
 ) -> Generator[None, MagicMock, None]:
     """Return a mocked Elgato client."""
+    fixtures = __load_fixtures(device_fixtures, state_variant)
     with patch(
         "homeassistant.components.elgato.coordinator.Elgato", autospec=True
     ) as elgato_mock:
         elgato = elgato_mock.return_value
-        elgato.info.return_value = Info.parse_raw(
-            load_fixture(f"{device_fixtures}/info.json", DOMAIN)
-        )
-        elgato.state.return_value = State.parse_raw(
-            load_fixture(f"{device_fixtures}/{state_variant}.json", DOMAIN)
-        )
-        elgato.settings.return_value = Settings.parse_raw(
-            load_fixture(f"{device_fixtures}/settings.json", DOMAIN)
-        )
+        elgato.info.return_value = fixtures[0]
+        elgato.state.return_value = fixtures[1]
+        elgato.settings.return_value = fixtures[2]
         yield elgato
```

After:

![image](https://user-images.githubusercontent.com/195327/217274713-c05fe2aa-1613-4ec6-81b4-0ba36e8933f9.png)

It is a tiny bit slower, I guess since I'm using NVMe there is not much to win? Is that worth the complexity?

../Frenck

Reply:
LRU added to our `load_fixture` helper in #87634